<launch>
  <arg name="map_path" default="$(find-pkg-share conveyor_utils)/launch/lab.yaml"/>

  <arg name="belt_speed" default="0.1"/>
  <arg name="package_length" default="0.5"/>
  <arg name="n_packs" default="6"/>
  <arg name="n_reds" default="3"/>

  <arg name="n_runs" default="5"/>
  <arg name="update_rate" default="10."/>

  <arg name="log" default="false"/>
  <arg name="log_rate" default="10."/>
  <arg name="exp_name" default=""/>
  <arg name="log_path" default="/home/gabri/Documents/logs_iros_test/"/>
  <arg name="interface" default="GUI"/>

  <arg name="joy_dev" default="0"/>

  <node name="pack_generator" pkg="iros" exec="pack_generator" output="screen" args="--ros-args --log-level info">
    <param name="belt_speed" value="$(var belt_speed)"/>
    <param name="package_length" value="$(var package_length)"/>
    <param name="n_packs" value="$(var n_packs)"/>
    <param name="n_reds" value="$(var n_reds)"/>
  </node>

  <node name="gui_experiment" pkg="iros" exec="gui_experiment" output="screen" args="--ros-args --log-level info">
    <param name="n_runs" value="$(var n_runs)"/>
    <param name="update_rate" value="$(var update_rate)"/>
  </node>

  <include file="$(find-pkg-share conveyor_utils)/launch/lab.launch">
    <arg name="map_path" value="$(var map_path)" />

    <arg name="belt_speed" value="$(var belt_speed)" />
    <arg name="package_length" value="$(var package_length)" />
    <arg name="joy_dev" value="$(var joy_dev)" />
  </include>

  <node name="gui" pkg="conveyor_gui" exec="conveyor_gui" output="screen" args="--ros-args --log-level info">
    <param name="map_path" value="$(var map_path)"/>
    <param name="orientation" value="1"/>
  </node>

  <node name="task_logger" pkg="iros" exec="task_logger" output="screen" args="--ros-args --log-level info" if="$(var log)">
    <param name="log_rate" value="$(var log_rate)"/>
    <param name="exp_name" value="$(var exp_name)"/>
    <param name="log_path" value="$(var log_path)"/>

    <param name="map_path" value="$(var map_path)"/>
    <param name="belt_speed" value="$(var belt_speed)"/>

    <param name="package_length" value="$(var package_length)"/>
    <param name="n_packs" value="$(var n_packs)"/>
    <param name="n_reds" value="$(var n_reds)"/>
    <param name="n_runs" value="$(var n_runs)"/>
    <param name="interface" value="$(var interface)"/>
    <param name="update_rate" value="$(var update_rate)"/>
  </node>

</launch>
